packages/ra-ui-materialui/src/layout/Sidebar.tsx

```diff
@@ -45,22 +55,21 @@ const useStyles = makeStyles(
     { name: 'RaSidebar' }
 );
 
-const Sidebar = props => {
-    const {
-        children,
-        closedSize,
-        size,
-        classes: classesOverride,
-        ...rest
-    } = props;
+interface Props extends DrawerProps {
+    children: ReactElement;
+}
+
+const Sidebar: FC<Props> = props => {
+    const { children, ...rest } = props;
     const dispatch = useDispatch();
-    const isXSmall = useMediaQuery(theme => theme.breakpoints.down('xs'));
-    const isSmall = useMediaQuery(theme => theme.breakpoints.down('sm'));
-    const open = useSelector(state => state.admin.ui.sidebarOpen);
-    useSelector(state => state.locale); // force redraw on locale change
+    const isXSmall = useMediaQuery<Theme>(theme =>
+        theme.breakpoints.down('xs')
+    );
+    const isSmall = useMediaQuery<Theme>(theme => theme.breakpoints.down('sm'));
+    const open = useSelector((state: ReduxState) => state.admin.ui.sidebarOpen);
     const handleClose = () => dispatch(setSidebarVisibility(false));
     const toggleSidebar = () => dispatch(setSidebarVisibility(!open));
-    const classes = useStyles({ ...props, open });
+    const classes = useStyles({ open });
```

Comment:
not good, we must pass the complete `props` object or this will cause rerenderings